src/libaktualizr/http/httpclient.cc

```diff
@@ -168,27 +168,22 @@ void HttpClient::setCerts(const std::string& ca, CryptoSource ca_source, const s
 }
 
 HttpResponse HttpClient::post(const std::string& url, const Json::Value& data) {
-  curlEasySetoptWrapper(curl, CURLOPT_URL, url.c_str());
-  curlEasySetoptWrapper(curl, CURLOPT_POST, 1);
+  CURL* curl_post = Utils::curlDupHandleWrapper(curl, pkcs11_key);
+  curlEasySetoptWrapper(curl_post, CURLOPT_URL, url.c_str());
+  curlEasySetoptWrapper(curl_post, CURLOPT_POST, 1);
+  if (pkcs11_key) {
```

Comment:
Should be unneeded here (in Utils::curlDupHandleWrapper)

Reply:
Yep, that was a mistake. Fixed.